app/controllers/users_controller.rb

```diff
@@ -1,29 +1,71 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: %i(index edit update destroy)
+  before_action :correct_user, only: %i(show edit destroy update)
+  before_action :admin_user, only: :destroy
+
+  def index
+    @pagy, @users = pagy User.by_earliest_created,
+                         items: Settings.digits.length_30
+  end
+
   def new
     @user = User.new
   end
 
-  def show
-    @user = User.find_by id: params[:id]
-    redirect_to :root, flash: {warning: t(".user_not_found")} if @user.nil?
-  end
+  def show; end
 
   def create
     @user = User.new user_params
     if @user.save
       flash[:success] = I18n.t "users.create.success"
       reset_session
       log_in @user
+      flash[:success] = I18n.t "users.create.success"
       redirect_to @user
     else
       render :new, status: :unprocessable_entity
     end
   end
 
+  def edit; end
+
+  def update
+    if @user.update user_params
+      flash[:success] = I18n.t "users.update.success"
+      redirect_to @user
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @user.destroy
+    if @user.destroy
+      flash[:success] = I18n.t "users.delete.success"
+      redirect_to users_url, status: :see_other
+    else
+      flash[:message] = I18n.t "users.destroy.recordCanNotDestroy"
```

Comment:
```suggestion
      flash[:danger] = I18n.t "users.destroy.recordCanNotDestroy"
```
lỗi xong rồi thì redirect đi đâu?